DEVDOCS-2531-remove-coupon-array-under-line-items

traci707 opened the pull request:
[DEVDOCS-2531](https://jira.bigcommerce.com/browse/DEVDOCS-2531)
Removed coupon array under cart physical and digital line items to correct the Response schema

Two changes:
1. Removed lines 1058 to 1079 (array) and replaced with line 1030 (object).
2. Removed lines 1212 to 1233 (array) and replaced with line 1153 (object).

capsula4 reviewed (approved):
👏

ektravel commented on reference/checkouts.v3.yml:
```diff
@@ -1048,35 +1022,12 @@ definitions:
                               format: double
                       coupons:
                         description: ''
-                        type: array
```
I believe this should be `type: object`. The AppliedCoupon model has all of the applicable properties: id, code, discounted_amount, coupon_type.

ektravel commented on reference/checkouts.v3.yml:
```diff
@@ -1202,35 +1145,12 @@ definitions:
-                              format: double
-                          required:
-                            - code
+                        type: number
```
I believe this should be `type: object`. The AppliedCoupon model has all of the applicable properties: id, code, discounted_amount, coupon_type.

ektravel reviewed (commented):
Added my comments.

traci707 commented:
Reviewed server-to-server carts and storefront checkouts as well. I made the following changes:
Storefront:
In Checkout-Resp, 
- changed "TaxIncluded" to "isTaxIncluded"
- added "isMutable", but I do not know what to put for definition.
In Checkout-Full,
- added "comparisonPrice"
- added "extendedComparisonPrice"

Server-to-server:
Physical, digital, gift certificates, and custom_items are good now. The array has been removed.

ektravel commented on reference/checkouts.sf.yml:
```diff
@@ -3609,6 +3612,9 @@ definitions:
                             format: double
+                          brand:
+                            description:
+                            type: 
```
```suggestion
                            type: string
```

ektravel commented on reference/checkouts.sf.yml:
```diff
@@ -3609,6 +3612,9 @@ definitions:
                             type: number
                             format: double
+                          brand:
+                            description:
```
```suggestion
                            description: The product brand.
```

ektravel reviewed (commented):
Added description and type for `brand` property.

aglensmith commented:
Closing due to conflicts. Reopen or push once conflicts are resolved.
